source/core_mqtt_prop_deserializer.c

```diff
@@ -278,7 +278,7 @@ MQTTStatus_t MQTT_GetNextPropertyType( MQTTPropBuilder_t * pPropertyBuilder,
 
     if( status != MQTTSuccess )
     {
-        LogError( ( "Property type is invalid." ) );
+        /* Do nothing. checkPropBuilderParams will log the warning/error. */
```

Comment:
Why not get rid of the first condition?

```suggestion
        if ( property == NULL )
        .....//rest of code
```

Reply:
In this case if the function `checkPropBuilderParams` returns an error, we'd essentially be ignoring it. We want to exit if the above function fails.